SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java

```diff
@@ -514,17 +1437,114 @@ public void testDoy() {
+    //ERROR:  unit "timezone_m" not supported for type date
+    //SELECT EXTRACT(TIMEZONE_H    FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone_h" not supported for type date
+
     @Test(expected = RuntimeException.class)
     // Calcite does not seem to know about DATE_TRUNC
     public void testDateTrunc() {
         String query = "SELECT DATE_TRUNC('MILLENNIUM', DATE '1970-03-20')";
         this.testQueryTwice(query, 0);
     }
 
+    // SELECT DATE_TRUNC('MILLENNIUM', DATE '1970-03-20'); -- 1001-01-01
+    //          date_trunc
+    //------------------------------
+    // Thu Jan 01 00:00:00 1001 PST
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('CENTURY', TIMESTAMP '1970-03-20 04:30:00.00000'); -- 1901
+    //        date_trunc
+    //--------------------------
+    // Tue Jan 01 00:00:00 1901
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('CENTURY', DATE '1970-03-20'); -- 1901
+    //          date_trunc
+    //------------------------------
+    // Tue Jan 01 00:00:00 1901 PST
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('CENTURY', DATE '2004-08-10'); -- 2001-01-01
+    //          date_trunc
+    //------------------------------
+    // Mon Jan 01 00:00:00 2001 PST
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('CENTURY', DATE '0002-02-04'); -- 0001-01-01
+    //          date_trunc
+    //------------------------------
+    // Mon Jan 01 00:00:00 0001 PST
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('CENTURY', DATE '0055-08-10 BC'); -- 0100-01-01 BC
+    //           date_trunc
+    //---------------------------------
+    // Tue Jan 01 00:00:00 0100 PST BC
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('DECADE', DATE '1993-12-25'); -- 1990-01-01
+    //          date_trunc
+    //------------------------------
+    // Mon Jan 01 00:00:00 1990 PST
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('DECADE', DATE '0004-12-25'); -- 0001-01-01 BC
+    //           date_trunc
+    //---------------------------------
+    // Sat Jan 01 00:00:00 0001 PST BC
+    //(1 row)
+    //
+    //SELECT DATE_TRUNC('DECADE', DATE '0002-12-31 BC'); -- 0011-01-01 BC
+    //           date_trunc
+    //---------------------------------
+    // Mon Jan 01 00:00:00 0011 PST BC
+
+    // select 'infinity'::date > 'today'::date as t;
+    // t
+    //---
+    // t
+    //(1 row)
+    //
+    //select '-infinity'::date < 'today'::date as t;
+    // t
+    //---
+    // t
+    //(1 row)
+    //
+    //select isfinite('infinity'::date), isfinite('-infinity'::date), isfinite('today'::date);
+    // isfinite | isfinite | isfinite
+    //----------+----------+----------
+    // f        | f        | t
+    //(1 row)
+    //
+    //select 'infinity'::date = '+infinity'::date as t;
+    // t
+    //---
+    // t
+
     @Test(expected = RuntimeException.class)
     // Calcite does not seem to support infinity dates
     public void testInfinity() {
         String query = "SELECT DATE 'infinity'";
```

Comment:
Anything we can do about this?

Reply:
We can do many things, the main question is whether it's worth our time or there are more useful things to do. I think infinity times can wait.